subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/listener/TestMetricsSender.kt

```diff
@@ -0,0 +1,66 @@
+package com.avito.runner.service.listener
+
+import com.avito.android.stats.GaugeMetric
+import com.avito.android.stats.StatsDSender
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+import com.avito.runner.service.model.TestCase
+import com.avito.runner.service.model.TestCaseRun
+import com.avito.runner.service.worker.device.Device
+import com.avito.time.TimeProvider
+
```

Comment:
Describe please the concept. It's harder to get the idea from implementation details.

Reply:
I tried and it becomes implementation details ))

- `DeviceWorker` calls `TestListener.intended()`
- `TestMetrics` saves this `intended` timestamp by key (testcase, executionumber)
- DeviceWorker` calls `TestListener.started()`
- `TestMetrics` saves this `started` timestamp by key (testcase, executionumber)
- `TestMetrics` sends delta between now and intended
- DeviceWorker` calls `TestListener.finished()`
- `TestMetrics` sends delta between now and started

Reply:
Thanks, I meant only part about metrics layout.